### PR TITLE
PIE-2393: Update flaky test api docs with optional query param

### PR DESCRIPTION
**Pull request by KatieWright26**
### Description

We are allowing users to filter their flaky test results based on name, scope and location. This PR updates the docs to reflect this change.

It is blocked until release.

Screenshot 2024-03-20 at 12 15 51 (2)

- [Sister PR](https://github.com/buildkite/buildkite/pull/15992)

### CODE REVIEW

File: pages/apis/rest_api/analytics/_flaky_tests_query_strings.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<table>
+<tbody>
+  <tr>
+    <th>
+      <code>search</code>
+    </th>
+    <td>
+      <span>Returns flaky tests with a <code>name</code> or <code>scope</code> that matches the search string. Users with the <a href="https://buildkite.com/docs/test-analytics/ruby-collectors">Ruby test collector</a> installed can also filter results by <code>location</code>.</span>
```
Comment (meghan-kradolfer) [inline pages/apis/rest_api/analytics/_flaky_tests_query_strings.md:8]: Should it be 'contains' rather than 'matches'?

**Review by meghan-kradolfer — APPROVED** [review]

**Review by jasmine-q — APPROVED** [review]

**Comment by jasmine-q** [comment]
@KatieWright26 should this have the "hold for next release" label?

**Comment by meghan-kradolfer** [comment]
Will you add the updated screenshots for the index's to this PR, or should we open a new one?

**Comment by KatieWright26** [comment]
> Will you add the updated screenshots for the index's to this PR, or should we open a new one?

I can just add them to this one

**Pull request merged**